### ISSUE TITLE
Module not found: Can't resolve 'redux-persist/constants'

### ISSUE THREAD

**Issue by jeznag** [awslabs/aws-mobile-appsync-sdk-js#410]
>  Note: If your issue/feature-request/question is regarding the AWS AppSync service, please log it in the 
  [official AWS AppSync forum](https://forums.aws.amazon.com/forum.jspa?forumID=280&start=0)

**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**

./node_modules/aws-appsync/lib/store.js
Module not found: Can't resolve 'redux-persist/constants' in '/Users/jeremynagel/dev/personal/recycling-hero/node_modules/aws-appsync/lib'

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**

I installed the latest version of app-sync and am getting the above error.

```
import AWSAppSyncClient from 'aws-appsync';

import { APP_SYNC_REGION, APP_SYNC_URI, API_KEY } from './config';

export const client = new AWSAppSyncClient({
  url: APP_SYNC_URI,
  region: APP_SYNC_REGION,
  auth: {
    type: 'API_KEY',
    apiKey: API_KEY,
  },
  disableOffline: true
});

// ...
<ApolloProvider client={client as any}>
        <Rehydrated>
          ...
        </Rehydrated>
</ApolloProvider>
```
**What is the expected behavior?**

No errors

**Which versions and which environment (browser, react-native, nodejs) / OS are affected by this issue? Did this work in previous versions?**
```
 "aws-appsync": "^1.8.0",
    "aws-appsync-react": "^1.2.8",
```

**Comment by c-ancia** [comment]
Hi,

I just updated the 2 packages aswell and get the same error now.

If it can help you, I fixed it by changing in aws-persist/lib/store.js line 68 
var constants_2 = require("redux-persist/constants");
by 
var constants_2 = require("redux-persist/lib/constants");

**Comment by OlivierPT** [comment]
Hi,

I get exactly the same when I try to use localforage as offline storage.
The local modification above fix the issue for me also.

**Comment by c-ancia** [comment]
Hi,

Yes it works, but just a little reminder that it's usually not best practice or a good idea to modify directly a dependency. I've wrote this down just in case and also hoping that maybe the maintainers would see it and add it to a future release!

**Comment by OlivierPT** [comment]
Exactly! That's why I just did a pull request with the expected change. I hope they include it quickly. That's really a shame they are not more reactive on this lib!

**Comment by c-ancia** [comment]
Oh okay, that's great, I hadn't noticed the pull request ahah, my bad! :)

**Comment by jordanranz** [comment]
Apologies for the delayed response, I made a comment on that PR.

**Comment by zootopia106** [comment]
Is this issue fixed by aws appsync team??

**Comment by OlivierPT** [comment]
There is a PR to fix it that is still waiting for validation... :(

**Comment by zootopia106** [comment]
It's taking too long. PR is made 2 weeks ago, I guess??

**Comment by jhleath** [comment]
There is a workaround if you are impatient for the fix to be merged in, I used the npm install at version to get both v4 and v5 of redux-persist in my app:

```
npm install --save redux-persist5@npm:redux-persist
npm install --save redux-persist@4
````

Then, in *your* application, do like:

```
import { persistStore, persistReducer } from 'redux-persist5';
```

which lets the AWS framework continue to use the `redux-persist` moniker for v4 and import correctly

**Comment by Zbish** [comment]
what about the fix for this ???
still waiting ??

**Comment by davidfarinha** [comment]
Bump. Is there any update on this?

**Comment by sankar9659** [comment]
Getting same issue, please give an solution

**Comment by Blind12** [comment]
Any news on this ?

**Comment by RaulTsc** [comment]
Any updates?

**Comment by OlivierPT** [comment]
Since AWS Team didn't reply in time, I didn't follow the topic (because I moved to other topics) and my PR, sorry.
It's more than 1 year now so I hope the issue has been fixed directly in the newer versions... isn't it?

**Comment by raaone7** [comment]
+1 , please update on the eta

**Comment by DIONNiklas** [comment]
This bug is still open!!

**Comment by Danue1** [comment]
Any news?

**Comment by tzutalin** [comment]
any update?

**Comment by andidev** [comment]
Can someone from AWS at least please comment? AppSync is a paid service. Tell us why we should use it if you just keep ignoring bugs like this? 
Upgrading react native has been a hell last 2 years. This bug makes it even a worse hell.

How come libs maintained by regular people can be better maintained then libs from companies for paid services? This is starting to get ridiculous. Facebook does the same thing for react native except thats not a paid service of course.

**Comment by dperconti** [comment]
Checking in to see if any updates have been made on this bug?

**Comment by jamesaucode** [comment]
The fix has been merged and will be included in the next release (usually on Thursday)

**Comment by andidev** [comment]
@jamesaucode superthanks! 
However there is actually another similar breaking dependency update needed. If you guys fix this too my hope for the future of development will come back 😂
There is a hack/workaround for that dependency issue but it involves switching to yarn and it does not work 100% of the times you run yarn install. So I am not sure how to tell all our developers to handle that project.

Its this issue
https://github.com/awslabs/aws-mobile-appsync-sdk-js/issues/654

pretty please take a look at that too ❤️

**Comment by jamesaucode** [comment]
@andidev absolutely! I will take a look at the issue you linked 😄

**Comment by KatyeStevens** [comment]
> 

When will the next release be available? It's been over a year.
